packages/plugins/documentation/server/services/documentation.js

```diff
@@ -133,17 +149,19 @@ module.exports = ({ strapi }) => {
         'full_documentation.json'
       );
 
-      const settings = _.cloneDeep(defaultConfig);
+      const settings = _.merge(_.cloneDeep(defaultConfig), this.getDocumentationSettings());
 
       const serverUrl = getAbsoluteServerUrl(strapi.config);
       const apiPath = strapi.config.get('api.rest.prefix');
 
-      _.set(settings, 'servers', [
-        {
-          url: `${serverUrl}${apiPath}`,
-          description: 'Development server',
-        },
-      ]);
+      if(_.isEmpty(settings.servers)) {
+        _.set(settings, 'servers', [
+          {
+            url: `${serverUrl}${apiPath}`,
+            description: 'Development server',
+          },
+        ]);
+      }
```

Comment:
I think we should just merge the custom settings after the default settings have been modified. 

See my [comment](https://github.com/strapi/strapi/pull/12070#pullrequestreview-871703431) on https://github.com/strapi/strapi/pull/12070#pullrequestreview-871703431

Reply:
Hi @markkaylor, I've updated the code to match with your comment on the other PR. Can you verify if this works for you guys?